### ISSUE TITLE
Sonar link in README is invalid

### ISSUE THREAD

**Issue by buildscientist** [greenmail-mail-test/greenmail#215]
The Sonar link referenced in the README is invalid. 

SonarCloud has introduced [GitHub integration](https://about.sonarcloud.io/get-started/) which requires auth tokens to be generated for a project. 

Sonar analysis hasn't been run against GreenMail since 9/2016 - see [Sonar dashboard](https://sonarcloud.io/dashboard?id=com.icegreen%3Agreenmail-parent) for more details.

**Comment by buildscientist** [comment]
@marcelmay @camann9 
Looks like the latest version of the Sonar Mavan plugin only support JDK 8.0 or above. 

When I run a CircleCI build with mvn sonar:sonar phase I get the following error: 

```Java
Caused by: java.lang.UnsupportedClassVersionError: org/sonarsource/scanner/maven/SonarQubeMojo : Unsupported major.minor version 52.0
```
See [build 113](https://circleci.com/gh/buildscientist/greenmail/113#queue-placeholder/containers/0) off of my branch for more details. 

Do we need to continue supporting JDK7 builds? Can someone get some stats from Maven central to see if we still have downloads for JDK7 Greenmail builds?

**Comment by marcelmay** [comment]
In general we support JDK 7 with GreenMail 1.5.x, therefore a JDK 7 build is essential.
We won't see certain issues (compile for 7, but accidentally use JDK 8 API ... runtime behaviour etc.).

The sonar analysis step can run any JDK >= 7, as it is the analysis part afterwards (no compilation, no artifacts creation, no test invocations).

So we could run sonar report regardless of the project JDK (in term of higher version), eg using only JDK 8. The disadvantage is a slightly more complex build, but we catch JDK 7 issues in GreenMail - and the build is automated anyway.

**Comment by buildscientist** [comment]
@marcelmay 

Makes sense - I wanted to avoid having a more complex build. I'll modify the build process to only run Sonar if we're building against JDK8 or above. Unfortunate that SonarQube is requiring JDK8 or above just to use their Maven plugin.

**Comment by buildscientist** [comment]
The [prerequisites](https://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner+for+Maven) from SonarQube's documentation stating that Java8 is required to use the Maven plugin.
